test/test_nn_syntax.js

```diff
@@ -618,6 +618,17 @@ now => @org.schema.restaurant() => notify;`],
      `get sunrise sunset on the NUMBER_0 th`, { NUMBER_0: 25 },
      `now => @org.thingpedia.weather.sunrise(date=new DatePiece(-1, -1, 25, -1)) => notify;`],
 
+    [`now => @org.thingpedia.weather.sunrise param:date:Date = new DatePiece ( - 1 , - 1 , 10 , TIME_0 ) => notify`,
```

Comment:
do we have to use space between `-` and `1` here?  It looks weird to me

Reply:
From what I could tell, negative numbers in NN Syntax are always separated from the `-`, such as [in here](https://github.com/stanford-oval/thingtalk/blob/master/lib/nn-syntax/tonn_converter.js#L264), so I kept it like that. Don't know if there's a special reason for it.
Cc @gcampax

Reply:
l talked to giovanni. let's not put -1 in the syntax.

Reply:
let's do `now => @org.thingpedia.weather.sunrise param:date:Date = new DatePiece ( , , 10 , TIME_0 ) => notify`

Reply:
or `now => @org.thingpedia.weather.sunrise param:date:Date = new DatePiece ( null , null , 10 , TIME_0 ) => notify` if you prefer

Reply:
Ok, as per my last commit now, the NN Syntax omits the non-supplied values (`new DatePiece ( , , 10 , TIME_0 )`).

In the ThingTalk and JS syntaxes, it's taking in `null` instead of `-1` for non-supplied (`new DatePiece(null, null, 10, time)`), But thinking now, maybe I misinterpreted and you had meant for these to stay the same with `-1`. Is it a bad thing that accidental `null`s would get interpreted the same as the intentional non-supplied args? Would be very quick to rollback this change, so let me know and I'll do it immediately.

Reply:
Oh, my take is there is no such thing as "accidental `null`". If "null" is not a valid value in a field, it's a programmer error (and we can put an assertion to catch it). If "null" is a valid value, we give it a meaning.

Reply:
Also, please make ThingTalk surface syntax consistent with NN syntax, unless there is a good reason not to. I don't care about consistency with JS, cause we can't change JS, obv.

Reply:
Updated ThingTalk surface syntax for consistency with NN syntax (e.g. `new DatePiece(, , 10, time)`)